src/common/health.rs

```diff
@@ -153,33 +153,33 @@ impl Task {
         // This allows to check the happy path without waiting for the first call.
         self.check_ready_signal.notify_one();
 
-        // Get *cluster* commit index, or check if this is the only node in the cluster
-        let Some(cluster_commit_index) = self.cluster_commit_index().await else {
+        // Get estimate of current cluster commit so we can wait for it
+        let Some(mut cluster_commit_index) = self.cluster_commit_index(true).await else {
```

Comment:
Minor change: the first consensus commit we wait on is just requested from a single node, which may not necessarily be the latest. It is fine because we'll always do another set of requests to all nodes later.

In an ideal situation this reduces requests to other nodes by (50% - 1).